android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/AddonManager.kt

```diff
@@ -98,12 +98,14 @@ class AddonManager(
             // that, we add the installed state on each add-on instance.
             //
             // NB: We're keeping translations only for the default locale.
-            val installedAddonGUIDs = otherInstalledExtensions
+            val installedAddons = otherInstalledExtensions
                 .filterValues { it.getMetadata()?.temporary == false }
-                .map { extensionEntry -> extensionEntry.value.id }
-            val installedAddons = addonsProvider.getAddonsByGUIDs(installedAddonGUIDs, language = userLanguage)
-                .map { addon -> addon.filterTranslations(locales) }
-                .map { addon -> addon.copy(installedState = installedExtensions[addon.id]?.toInstalledState()) }
+                .map { ext ->
+                    Addon.newFromWebExtension(
+                        ext.value,
+                        installedExtensions[ext.key]?.toInstalledState(),
+                    )
+                }
```

Comment:
Noticed you where catching if we had an issues fetching `getFeaturedAddons`. but that will make us show an partial list of add-ons, right? It won't be better to continue following the same pattern as we havenow, where if we have an error loading parts of the whole list of add-on, we show an error message and users report it. If we show a partial list we may be hiding issues and user not reporting them?

```kotlin
var featuredAddons = emptyList<Addon>()
            try {
                val userLanguage = Locale.getDefault().language
                val locales = listOf(userLanguage)
                featuredAddons = addonsProvider.getFeaturedAddons(allowCache, language = userLanguage)
                    .filter { addon -> !installedExtensions.containsKey(addon.id) }
                    .map { addon -> addon.filterTranslations(locales) }
            } catch (throwable: Throwable) {
                // Do not throw when we fail to fetch the featured add-ons since there can be installed add-ons.
                logger.warn("Failed to get the featured add-ons", throwable)
            }
```

Reply:
There are 2 types of add-ons we show in this list:

1. installed extensions: do not require any API call anymore (with this patch) and very important to show in the list. We must always list them IMO
2. featured add-ons: fetched from AMO, nice to have but not required since we now have a button to open AMO in this list

Reply:
Thanks for adding more context!

Reply:
Lets follow the same behaviour as desktop!